client/virtual_channel.go

```diff
@@ -100,6 +100,52 @@ func (c *Client) handleVirtualChannelFundingProposal(
 	c.acceptProposal(responder)
 }
 
+func (c *Channel) watchVirtual() error {
+	log := c.Log().WithField("proc", fmt.Sprintf("virtual channel watcher %v", c.ID()))
+	defer log.Info("Watcher returned.")
+
+	// Subscribe to state changes
+	ctx := c.Ctx()
+	sub, err := c.adjudicator.Subscribe(ctx, c.Params())
+	if err != nil {
+		return errors.WithMessage(err, "subscribing to adjudicator state changes")
+	}
+	defer func() {
+		err := sub.Close()
+		if err != nil {
+			log.Warn(err)
+		}
+	}()
+
+	// Wait for state changed event
+	for e := sub.Next(); e != nil; e = sub.Next() {
+		log.Debugf("event %v", e)
+
+		// Update channel
+		switch e := e.(type) {
+		case *channel.RegisteredEvent:
```

Comment:
observation:
The virtual channel may be registered with sub-channels.

problem:
the contract requires all sub-channel states for registration, but the hub currently doesn't persist them.
hence, the hub cannot conclude the mirror channel tree.

solution:
- option1 (low effort)
  - don't allow virtual channels with subchannels
- option2 (medium effort)
  - on registration of ledger channel, persist all sub-channels
  - only ledger channel watching required
  - extend persistence

Reply:
Going with option1 for now